fabric-convention-tags-v2/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.convention.datagen.generators;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.registry.RegistryWrapper;
+import net.minecraft.registry.tag.BlockTags;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataOutput;
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricTagProvider;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalBlockTags;
+import net.fabricmc.fabric.api.tag.convention.v2.TagUtil;
+
+public final class BlockTagGenerator extends FabricTagProvider.BlockTagProvider {
+	static List<Block> VILLAGER_JOB_SITE_BLOCKS = List.of(
+			Blocks.BARREL,
+			Blocks.BLAST_FURNACE,
+			Blocks.BREWING_STAND,
+			Blocks.CARTOGRAPHY_TABLE,
+			Blocks.CAULDRON,
+			Blocks.LAVA_CAULDRON,
+			Blocks.WATER_CAULDRON,
+			Blocks.POWDER_SNOW_CAULDRON,
+			Blocks.COMPOSTER,
+			Blocks.FLETCHING_TABLE,
+			Blocks.GRINDSTONE,
+			Blocks.LECTERN,
+			Blocks.LOOM,
+			Blocks.SMITHING_TABLE,
+			Blocks.SMOKER,
+			Blocks.STONECUTTER
+	);
+
+	public BlockTagGenerator(FabricDataOutput output, CompletableFuture<RegistryWrapper.WrapperLookup> registriesFuture) {
+		super(output, registriesFuture);
+	}
+
+	@Override
+	protected void configure(RegistryWrapper.WrapperLookup registries) {
+		getOrCreateTagBuilder(ConventionalBlockTags.STONES)
+				.add(Blocks.ANDESITE)
```

Comment:
Normal stone (and probably deepslate) should come first in the stone tag, it is far more representative than andesite.

Reply:
Stone definitely but it seems vanilla sticks deepslate to last. I'll match vanilla for now until I figure out what I am ultimately gonna do with this block stone tag since vanilla's exist
https://github.com/misode/mcmeta/blob/data-json/data/minecraft/tags/blocks/base_stone_overworld.json